python/semantic_kernel/connectors/memory/weaviate/utils.py

```diff
@@ -256,43 +246,46 @@ def extract_key_from_weaviate_object_based_on_data_model_definition(
 def extract_vectors_from_weaviate_object_based_on_data_model_definition(
     weaviate_object,
     data_model_definition: VectorStoreRecordDefinition,
+    named_vectors: bool,
 ) -> dict[str, Any]:
     """Extract vectors from a Weaviate object based on the data model definition.
 
-    Named vectors use the names of the associated data fields.
-    Rest of the vectors use their own names.
-
-    Expecting the Weaviate object to have all the vectors defined in the data model definition.
-
     Args:
         weaviate_object: The Weaviate object.
         data_model_definition (VectorStoreRecordDefinition): The data model definition.
+        named_vectors (bool): Whether the collection uses named vectors.
 
     Returns:
-        dict[str, Any]: The vectors.
+        dict[str, Any]: The vectors, or None.
     """
-    # Named vectors: key is the data field name, value is the vector field name
-    named_vectors: dict[str, str] = {
-        field.name: field.embedding_property_name
-        for field in data_model_definition.fields.values()
-        if isinstance(field, VectorStoreRecordDataField) and field.has_embedding
-    }
-    # Unnamed vectors: a list of vector fields that are not associated with a data field
-    unnamed_vectors: list[str] = [
-        field.name
-        for field in data_model_definition.fields.values()
-        if isinstance(field, VectorStoreRecordVectorField)
-        and not field.name
-        and field.name not in named_vectors.values()
-    ]
-
-    # Combine named and unnamed vectors: Unnamed vectors will use their own names
-    vectors: dict[str, str] = named_vectors | {vector_name: vector_name for vector_name in unnamed_vectors}
-
-    return {
-        vector_field_name: weaviate_object.vector[data_field_name]
-        for data_field_name, vector_field_name in vectors.items()
-    }
+    if not weaviate_object.vector:
+        return {}
+    if named_vectors:
+        return {
+            vector.name: weaviate_object.vector[vector.name]
+            for vector in data_model_definition.vector_fields
+            if vector.name in weaviate_object.vector
+        }
+    vector_field = data_model_definition.vector_fields[0] if data_model_definition.vector_fields else None
+    if not vector_field:
+        return {}
+    return {vector_field.name: weaviate_object.vector["default"]}
 
 
 # endregion
+# region VectorSearch helpers
+
+
+def _create_filter_from_vector_search_filters(filters: VectorSearchFilter | None) -> "_Filters | None":
+    if not filters:
+        return None
+    weaviate_filters: list["_Filters"] = []
+    for filter in filters.filters:
+        match filter:
+            case EqualTo():
```

Comment:
For my own education purposes: without override the compare methods, will these two objects be considered equal?

Reply:
not sure I follow, but EqualTo is a class, so this is the syntax instead of using if isinstance(filter, EqualTo): etc.

Reply:
`EqualTo()` creates an object tho.

Reply:
no, it looks like a constructor, but it isn't with a match: https://docs.python.org/3.10/tutorial/controlflow.html#match-statements

Reply:
Learned something new each day!